worker/cmd/generate.go

```diff
@@ -188,15 +191,6 @@ var generateCmd = &cobra.Command{
 			close(stopChan)
 		}(sigChan)
 
-		wg.Add(1)
-		go func(ch <-chan string) {
-			defer wg.Done()
-			for job := range ch {
-				jp := NewJobProcessor(ctx, pool, svc, sugar, job, PreCheckEndpointURL, SdgEndpointURL, TlsClientCertPath, TlsClientKeyPath, TlsServerCaCertPath)
-				jp.processJob()
-			}
-		}(jobChan)
-
```

Comment:
```go
...
                 jp.processJob()
                 jobInProgress = false
...

```

Reply:
This change sounds good to me as well. But i think still the worker won't exit till the processJob() is done with it work. I think we need to pass `signal.NotifyContext` to the Worker, so that it can abort whenever the signal is received, that will exit processJob() and eventually exist the worker as well. wdyt?

Reply:
@russellb added the NotifyContext and passed to Precheck/generate-local command execution as well as sdg post http request. That should abort the processJob() one os signal now, and worker should shutdown gracefully. If you still see any corner case where it can fail, let me know, i can update the pr with your suggestion.

Reply:
i think this is OK